Apply platform schema (#164)
fingerprintjs/fingerprint-pro-server-api-openapi: .changeset/quiet-parrots-stick.md

---
'fingerprint-pro-server-api-openapi': major
---

- Rename `BotdResult` -> `Botd`.
- Rename `BotdDetectionResult` -> `BotdBot`:
  - Extract `result` type as `BotdBotResult`.
- Rename `ClonedAppResult` -> `ClonedApp`.
- Rename `DeveloperToolsResult` -> `DeveloperTools`.
- Rename `EmulatorResult` -> `Emulator`.
- Refactor error models:
  - Remove `ErrorCommon403Response`, `ErrorCommon429Response`, `ErrorEvent404Response`, `TooManyRequestsResponse`, `ErrorVisits403`, `ErrorUpdateEvent400Response`, `ErrorUpdateEvent409Response`, `ErrorVisitor400Response`, `ErrorVisitor404Response`, `IdentificationError`, `ProductError`.
  - Introduce `ErrorResponse` and `ErrorPlainResponse`.
    - [note]: `ErrorPlainResponse` has a different format `{ "error": string }` and it is used only in `GET /visitors`.
  - Extract `error` type as `Error`.
  - Extract `error.code` type as `ErrorCode`.
- Rename `EventResponse` -> `EventsGetResponse`.
- Rename `EventUpdateRequest` -> `EventsUpdateRequest`.
- Rename `FactoryResetResult` -> `FactoryReset`.
- Rename `FridaResult` -> `Frida`.
- Rename `IPLocation` -> `Geolocation`:

>   - Extract `city` type as `GeolocationCity`.

Reviewer: ```suggestion
  - Rename `IPLocationCity` -> `GeolocationCity`.
```

  - Extract `subdivisions` type as `GeolocationSubdivisions`.
  - Rename `Location` -> `GeolocationContinent`:
  - Introduce a dedicated type `GeolocationCountry`.
  - Rename `Subdivision` -> `GeolocationSubdivision`.
- Rename `HighActivityResult` -> `HighActivity`.
- Rename `Confidence` -> `IdentificationConfidence`.
- Rename `SeenAt` -> `IdentificationSeenAt`.
- Rename `IncognitoResult` -> `Incognito`.
- Rename `IpBlockListResult` -> `IPBlocklist`:
  - Extract `details` type as `IPBlocklistDetails`.
- Rename `IpInfoResult` -> `IPInfo`:

>   - Extract `v4` type as `IPInfoV4`.
>   - Extract `v6` type as `IPInfoV6`.

Reviewer: ```suggestion
  - Rename `IpInfoResultV4` -> `IPInfoV4`.
  - Rename `IpInfoResultV6` -> `IPInfoV6`.
```

  - Rename `ASN` -> `IPInfoASN`.
  - Rename `DataCenter` -> `IPInfoDataCenter`.
- Rename `JailbrokenResult` -> `Jailbroken`.
- Rename `LocationSpoofingResult` -> `LocationSpoofing`.
- Rename `PrivacySettingsResult` -> `PrivacySettings`.

> - Rename `ProductsResponse` -> `Products`:
>   - Extract inner types: `ProductIdentification`, `ProductBotd`, `ProductRootApps`, `ProductEmulator`, `ProductIPInfo`, `ProductIPBlocklist`, `ProductTor`, `ProductVPN`, `ProductProxy`, `ProductIncognito`, `ProductTampering`, `ProductClonedApp`, `ProductFactoryReset`, `ProductJailbroken`, `ProductFrida`, `ProductPrivacySettings`, `ProductVirtualMachine`, `ProductRawDeviceAttributes`, `ProductHighActivity`, `ProductLocationSpoofing`, `ProductSuspectScore`, `ProductRemoteControl`, `ProductVelocity`, `ProductDeveloperTools`.

Reviewer: ```suggestion
- Rename `ProductsResponse` -> `Products`:
  - Rename inner types: `ProductsResponseIdentification` -> `ProductIdentification`, `ProductsResponseIdentificationData` -> `Identification`,  `ProductsResponseBotd` -> `ProductBotd`, `SignalResponseRootApps` -> `ProductRootApps`, `SignalResponseEmulator` -> `ProductEmulator`, `SignalResponseIpInfo` -> `ProductIPInfo`, `SignalResponseIpBlocklist` -> `ProductIPBlocklist`, `SignalResponseTor` -> `ProductTor`, `SignalResponseVpn` -> `ProductVPN`, `SignalResponseProxy` -> `ProductProxy`, `ProxyResult` -> `Proxy`, `SignalResponseIncognito` -> `ProductIncognito`, `SignalResponseTampering` -> `ProductTampering`, `SignalResponseClonedApp` -> `ProductClonedApp`, `SignalResponseFactoryReset` -> `ProductFactoryReset`, `SignalResponseJailbroken` -> `ProductJailbroken`, `SignalResponseFrida` -> `ProductFrida`, `SignalResponsePrivacySettings` -> `ProductPrivacySettings`, `SignalResponseVirtualMachine` -> `ProductVirtualMachine`, `SignalResponseRawDeviceAttributes` -> `ProductRawDeviceAttributes`, `RawDeviceAttributesResultValue` -> `RawDeviceAttributes`, `SignalResponseHighActivity` -> `ProductHighActivity`, `SignalResponseLocationSpoofing` -> `ProductLocationSpoofing`, `SignalResponseSuspectScore` -> `ProductSuspectScore`, `SignalResponseRemoteControl` -> `ProductRemoteControl`, `SignalResponseVelocity` -> `ProductVelocity`, `SignalResponseDeveloperTools` -> `ProductDeveloperTools`.
```

  - Extract `identification.data` type as `Identification`.
- Rename `RawDeviceAttributesResult` -> `RawDeviceAttributes`:
  - Extract item type as `RawDeviceAttribute`.
  - Extract `error` type as `RawDeviceAttributeError`.
- Rename `RemoteControlResult` -> `RemoteControl`.
- Rename `RootAppsResult` -> `RootApps`.
- Rename `SuspectScoreResult` -> `SuspectScore`.
- Extract new model `Tag`.
- Rename `TamperingResult` -> `Tampering`.
- Rename `TorResult` -> `Tor`.
- Rename `VelocityResult` -> `Velocity`:
  - Rename `VelocityIntervals` -> `VelocityData`.
  - Rename `VelocityIntervalResult` -> `VelocityIntervals`.
- Rename `VirtualMachineResult` -> `VirtualMachine`.

> - Extract the `Visit` field `ipLocation` type as `DeprecatedGeolocation`.

Reviewer: ```suggestion
- Rename the `Visit` field `ipLocation` type `DeprecatedIPLocation` -> `DeprecatedGeolocation`.
  - Instead of `DeprecatedIPLocationCity` use common `GeolocationCity`
```

> - Rename `Response` -> `VisitorsGetResponse`.

Reviewer: ```suggestion
- Rename `Response` -> `VisitorsGetResponse`.
  - Omit extra inner type `ResponseVisits`
```
